godot-core/src/meta/signature.rs

```diff
@@ -5,520 +5,371 @@
-                    )*
-                ];
-
-                let result = new_from_ptrcall::<Self::Ret>(|return_ptr| {
-                    utility_fn(return_ptr, arg_ptrs.as_ptr(), arg_ptrs.len() as i32);
-                });
-                result.unwrap_or_else(|err| return_error::<Self::Ret>(&call_ctx, err))
-            }
+    /// Make a ptrcall to the Godot engine for a builtin method.
+    ///
+    /// # Safety
+    ///
+    /// - `builtin_fn` must expect explicit args `args`, and return a value of type `Ret`
+    #[inline]
+    pub unsafe fn out_builtin_ptrcall(
+        builtin_fn: sys::BuiltinMethodBind,
+        // Separate parameters to reduce tokens in generated class API.
+        class_name: &'static str,
+        method_name: &'static str,
+        type_ptr: sys::GDExtensionTypePtr,
+        args: Params,
+    ) -> Ret {
+        let call_ctx = CallContext::outbound(class_name, method_name);
+        // $crate::out!("out_builtin_ptrcall: {call_ctx}");
+
+        unsafe {
+            Self::perform_ptrcall(args, &call_ctx, |explicit_args, return_ptr| {
+                builtin_fn(
+                    type_ptr,
+                    explicit_args.as_ptr(),
+                    return_ptr,
+                    explicit_args.len() as i32,
+                );
+            })
         }
-    };
-}
+    }
 
-/// Convert the `N`th argument of `args_ptr` into a value of type `P`.
-///
-/// # Safety
-/// - It must be safe to dereference the pointer at `args_ptr.offset(N)` .
-unsafe fn varcall_arg<P: FromGodot, const N: isize>(
-    args_ptr: *const sys::GDExtensionConstVariantPtr,
-    call_ctx: &CallContext,
-) -> Result<P, CallError> {
-    let variant_ref = Variant::borrow_var_sys(*args_ptr.offset(N));
-
-    P::try_from_variant(variant_ref)
-        .map_err(|err| CallError::failed_param_conversion::<P>(call_ctx, N, err))
+    /// Make a ptrcall to the Godot engine for a utility function.
+    ///
+    /// # Safety
+    ///
+    /// - `utility_fn` must expect explicit args `args`, and return a value of type `Ret`
+    #[inline]
+    pub unsafe fn out_utility_ptrcall(
+        utility_fn: sys::UtilityFunctionBind,
+        function_name: &'static str,
+        args: Params,
+    ) -> Ret {
+        let call_ctx = CallContext::outbound("", function_name);
+        // $crate::out!("out_utility_ptrcall: {call_ctx}");
+
+        unsafe {
+            Self::perform_ptrcall(args, &call_ctx, |explicit_args, return_ptr| {
+                utility_fn(
+                    return_ptr,
+                    explicit_args.as_ptr(),
+                    explicit_args.len() as i32,
+                );
+            })
+        }
+    }
+
+    /// Performs a ptrcall and processes the return value to give nice error output.
+    ///
+    /// # Safety
+    ///
+    /// This calls [`GodotFfi::new_with_init`] and passes the ptr as the second argument to `f`, see that function for safety docs.
+    unsafe fn perform_ptrcall(
```

Comment:
The brief description of the method is good, but "perform" doesn't carry any of that information. Is it maybe possible to make the name a bit more descriptive, even if longer?

Reply:
i can't really think of anything that's of similar length, but maybe just `perform_ptrcall_and_process_return` works?

Reply:
"process" isn't much better... I'd like to avoid generic filler words; if it's not easy to add more context, let's keep it short.

Maybe something highlighting that this operates on the raw FFI args/ret: ```raw_ptrcall(...)```
That would at least differentiate it from the other functions like `out_builtin_ptrcall` and so on.